File: RickyAI3_1.py
import json
import os
import requests

# File paths
json_data_path = "learning_bot_data.json"

# Function to train the bot with user input
def train_bot(input_text, responses):
    response = input(f"Enter the response for '{input_text}': ")
    responses[input_text] = response
    return responses

# Function to save the trained data to a JSON file
def save_bot_data(responses):
    with open(json_data_path, 'w') as file:
        json.dump(responses, file, indent=4)

# Function to load the trained data from a JSON file
def load_bot_data():
    responses = {}
    if os.path.exists(json_data_path):
        with open(json_data_path, 'r') as file:
            responses = json.load(file)
    return responses

# Function to perform a Google search
def google_search(query, api_key, cx):
    base_url = "https://www.googleapis.com/customsearch/v1"
    params = {
        'q': query,
        'key': api_key,
        'cx': cx,
    }

    response = requests.get(base_url, params=params)
    results = response.json()

    if 'items' in results and len(results['items']) > 0:
        top_result = results['items'][0]
        return f"Title: {top_result['title']}\nLink: {top_result['link']}\nSnippet: {top_result.get('snippet', 'N/A')}"
    else:
        return "No results found on Google."

# Load existing responses or start with an empty dictionary
bot_responses = load_bot_data()

# Google API key and custom search engine ID
google_api_key = 'AIzaSyAOll73Mr1n4FjdRKWP0hbNVO3hplCGoXM'
google_cx = '8007ac92de2d449f9'

# Chat with the bot
print("Welcome to RickyAI 3.1 Turbo Fly")
while True:
    user_input = input("You: ")
    if user_input.lower() in ['exit', 'stop']:
        break

    if user_input in bot_responses:
        response = bot_responses[user_input]
    else:
        print("RickyAIFramework: Searching Google")
        response = google_search(user_input, google_api_key, google_cx)
        print(f"Ricky: {response}")
        continue

    print(f"Ricky: {response}")

# Replace 'YOUR_API_KEY' and 'YOUR_CX' with your actual API key and custom search engine ID
api_key = 'AIzaSyAOll73Mr1n4FjdRKWP0hbNVO3hplCGoXM',
cx = '8007ac92de2d449f9'

File: RickyAI3_1.js
const fs = require('fs');
const readline = require('readline');
const axios = require('axios');

// File paths
const json_data_path = 'learning_bot_data.json';

// Function to train the bot with user input
async function trainBot(inputText, responses) {
  const response = await askQuestion(`Enter the response for '${inputText}': `);
  responses[inputText] = response;
  return responses;
}

// Function to save the trained data to a JSON file
function saveBotData(responses) {
  fs.writeFileSync(json_data_path, JSON.stringify(responses, null, 4));
}

// Function to load the trained data from a JSON file
function loadBotData() {
  let responses = {};
  if (fs.existsSync(json_data_path)) {
    const data = fs.readFileSync(json_data_path, 'utf-8');
    responses = JSON.parse(data);
  }
  return responses;
}

// Function to perform a Google search
async function googleSearch(query, apiKey, cx) {
  const base_url = 'https://www.googleapis.com/customsearch/v1';
  const params = {
    q: query,
    key: apiKey,
    cx: cx,
  };

  try {
    const response = await axios.get(base_url, { params });
    const results = response.data;

    if ('items' in results && results.items.length > 0) {
      const topResult = results.items[0];
      return `Title: ${topResult.title}\nLink: ${topResult.link}\nSnippet: ${topResult.snippet || 'N/A'}`;
    } else {
      return 'No results found on Google.';
    }
  } catch (error) {
    console.error('Error during Google search:', error.message);
    return 'Error during Google search.';
  }
}

// Function to ask a question and return the user's input
async function askQuestion(question) {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  return new Promise((resolve) => {
    rl.question(question, (answer) => {
      rl.close();
      resolve(answer);
    });
  });
}

// Load existing responses or start with an empty object
const botResponses = loadBotData();

// Google API key and custom search engine ID
const googleApiKey ='AIzaSyAOll73Mr1n4FjdRKWP0hbNVO3hplCGoXM';
const googleCx = '8007ac92de2d449f9';
// Chat with the bot
console.log('Welcome to RickyAI 3.1 Turbo Fly');
(async () => {
  while (true) {
    const userInput = await askQuestion('You: ');
    if (userInput.toLowerCase() === 'exit' || userInput.toLowerCase() === 'stop') {
      break;
    }

    let response;
    if (userInput in botResponses) {
      response = botResponses[userInput];
    } else {
      console.log('RickyAIFramework: Searching Google');
      response = await googleSearch(userInput, googleApiKey, googleCx);
      console.log(`Ricky: ${response}`);
      continue;
    }

    console.log(`Ricky: ${response}`);
  }
})();
